terasoluna-gfw-common-libraries/terasoluna-gfw-codepoints/src/test/java/org/terasoluna/gfw/common/codepoints/CodePointsTest.java

```diff
@@ -687,4 +690,33 @@ public void testEquals_same_codepoints_other_class() {
         assertThat(cp1.equals(cp2), is(false));
     }
 
+    @Test
+    public void testEqualsEmpty() {
+        // set up
+        CodePoints cp1 = new CodePoints("ABCD");
+        CodePoints cp2 = new CodePoints("    ");
+
+        // assert
+        assertThat(cp1.equals(cp2), is(false));
+    }
+
+    @Test
+    public void testEqualsNull() {
+        // set up
+        CodePoints cp1 = new CodePoints("ABCD");
+        String str = null;

```

Comment:
(re-add unintentionally removed comment)
- [ ] Should not you change class String to CodePoints?

Please look CodePoints#equals.
This code fulfills both conditions o == null and getClass() != o.getClass()...

----
It was modified.
13e4bad7fa41e61e98c8269d92c83091f5038b64